src/Symfony/Component/HttpClient/Response/AmpResponse.php

```diff
@@ -91,6 +91,28 @@ public function __construct(AmpClientState $multi, Request $request, array $opti
             return self::generateResponse($request, $multi, $id, $info, $headers, $canceller, $options, $onProgress, $handle, $logger);
         });
 
+        $info['pause_handler'] = static function (float $duration) use ($id, &$delay) {
+            if (null !== $delay) {
+                Loop::cancel($delay);
+                $delay = null;
+            }
+
+            if (0 < $duration) {
+                $duration += microtime(true);
+                Loop::disable($id);
+                $delay = Loop::defer(static function () use ($duration, $id, &$delay) {
+                    if (0 < $duration -= microtime(true)) {
+                        $delay = Loop::delay(ceil(1000 * $duration), static function () use ($id) { Loop::enable($id); });
+                    } else {
+                        $delay = null;
+                        Loop::enable($id);
+                    }
+                });
```

Comment:
PR ready.
Here is the workaround for amphp/amp#319
/cc @kelunik FYI

Reply:
@nicolas-grekas The Loop::defer should already be enough, no need to check the actually passed time.

Reply:
The time computation accounts for the duration of the tick itself. Since this delay is added at the end of the tick, isn't it?